feat(highlight): support highlight namespaces per window

bfredl opened the pull request:
The test will be if we can reimplement 'winhl' entirely lua side based on temas.

bfredl commented:
Demonstration (pick your two favorite color schemes):
```
local a = vim.api
_G.a = vim.api

vim.cmd [[ color wip ]]
atema = a.nvim_create_namespace("hösttema")
adata = a.nvim__get_hl_defs(0)
for k,v in pairs(adata) do
  a.nvim_set_hl(atema, k, v)
end

vim.cmd [[ color gruvbox ]]
btema = a.nvim_create_namespace("jultema")
bdata = a.nvim__get_hl_defs(0)
for k,v in pairs(bdata) do
  a.nvim_set_hl(btema, k, v)
end

vim.cmd [[ color wip ]]

a.nvim_win_set_config(0, {hl_ns=btema})
split
```

justinmk commented on src/nvim/api/vim.c:
```diff
@@ -250,32 +249,37 @@ void nvim_set_hl(Integer ns_id, String name, Dictionary val, Error *err)
+///
+/// @param ns_id the namespace to activate
+/// @param[out] err Error details, if any
+void nvim_set_hl_ns_fast(Integer ns_id, Error *err)
```
don't think we should start adding "fast" suffix to api names. It's already covered by an attribute and could be part of the metadata if it isn't already.

justinmk commented on src/nvim/api/vim.c:
```diff
@@ -250,32 +249,37 @@ void nvim_set_hl(Integer ns_id, String name, Dictionary val, Error *err)
 /// @param[out] err Error details, if any
-void nvim__set_hl_ns(Integer ns_id, Error *err)
-  FUNC_API_FAST
+void nvim_set_hl_ns(Integer ns_id, Error *err)
```
should we add a `opt` param now that this is promoted to public api?

bfredl committed:
feat(ui): allow to set the highlight namespace per window

- reimplement 'winhl' in terms of highlight namespaces
- check for EOF in screen tests (to indicate a likely crash)

luukvbaal commented:
https://github.com/neovim/neovim/issues/19823
